src/services/big-query-client.ts

```diff
@@ -35,4 +37,18 @@ export class HttpBigQueryClient extends HttpBaseClient implements BigQueryClient
       return { success: false, error: error };
     }
   }
+
+  public async saveEmailsData(payload: EmailList): Promise<SaveEmailsResult> {
+    const { jwtToken, email } = this.getApiConnectionData(this.clientName);
+    const response = await this.axios.post(`/big-query/${email}/save-emails`, {
```

Comment:
In my opinion, it should be a `PUT` against the same URL:

```suggestion
    const response = await this.axios.post(`/big-query/${email}/allowed-emails`, {
```

In my opinion, it is the same resource that you obtain with the `GET`, also, it is a kind of update, an idempotent operation, so `PUT` fits much better.

Reply:
suggest accepted and applied!!!